pkg/kopia/command/commands.go

```diff
@@ -0,0 +1,486 @@
+// Copyright 2022 The Kanister Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package command
+
+import (
+	"strconv"
+
+	"github.com/kanisterio/kanister/pkg/field"
+	"github.com/kanisterio/kanister/pkg/kopia"
+	"github.com/kanisterio/kanister/pkg/log"
+	"github.com/kanisterio/kanister/pkg/logsafe"
+	"github.com/kanisterio/kanister/pkg/utils"
+)
+
+func bashCommand(args logsafe.Cmd) []string {
+	log.Info().Print("Kopia Command", field.M{"Command": args.String()})
+	return []string{"bash", "-o", "errexit", "-c", args.PlainText()}
+}
+
+func stringSliceCommand(args logsafe.Cmd) []string {
+	log.Info().Print("Kopia Command", field.M{"Command": args.String()})
+	return args.StringSliceCMD()
+}
+
+func commonArgs(password, configFilePath, logDirectory string, requireInfoLevel bool) logsafe.Cmd {
+	c := logsafe.NewLoggable(kopiaCommand)
+	if requireInfoLevel {
+		c = c.AppendLoggable(logLevelInfoFlag)
+	} else {
+		c = c.AppendLoggable(logLevelErrorFlag)
+	}
+	if configFilePath != "" {
+		c = c.AppendLoggableKV(configFileFlag, configFilePath)
+	}
+	if logDirectory != "" {
+		c = c.AppendLoggableKV(logDirectoryFlag, logDirectory)
+	}
+	if password != "" {
+		c = c.AppendRedactedKV(passwordFlag, password)
+	}
+	return c
+}
+
+// ExecKopiaArgs returns the basic Argv for executing kopia with the given config file path.
+func ExecKopiaArgs(configFilePath string) []string {
+	return commonArgs("", configFilePath, "", false).StringSliceCMD()
+}
+
+// SnapshotCreateCommand returns the kopia command for creation of a snapshot
+// TODO: Have better mechanism to apply global flags
+func SnapshotCreateCommand(encryptionKey, pathToBackup, configFilePath, logDirectory string) []string {
```

Comment:
I was thinking should we drop `*Command` suffix from all the methods since the package name is also command. When it is imported and used, it will become `command.SnapshotCreate` which sounds better than `command.SnapshotCreateCommand`.

Reply:
This is done in PR C - https://github.com/kanisterio/kanister/pull/1552